portal/src/graphql/adminapi/RolesScreen.tsx

```diff
@@ -64,34 +86,83 @@ const RolesScreen: React.VFC = function RolesScreen() {
     }
     return encodeOffsetToCursor(offset - 1);
   }, [isSearch, offset]);
-  const { data, loading } = useQuery<
+
+  const onChangeOffset = useCallback((offset) => {
+    setOffset(offset);
+  }, []);
+
+  const onChangeSearchKeyword = useCallback((_e, value) => {
+    if (value != null) {
+      setSearchKeyword(value);
+      // Reset offset when search keyword was changed.
+      setOffset(0);
+    }
+  }, []);
+
+  const onClearSearchKeyword = useCallback((_e) => {
+    setSearchKeyword("");
+    // Reset offset when search keyword was changed.
+    setOffset(0);
+  }, []);
+
+  const { data, loading, previousData } = useQuery<
     RolesListQueryQuery,
     RolesListQueryQueryVariables
   >(RolesListQueryDocument, {
     variables: {
-      searchKeyword,
-      pageSize,
+      pageSize: isSearch ? searchResultSize : pageSize,
+      searchKeyword: debouncedSearchKey,
       cursor,
     },
     fetchPolicy: "network-only",
   });
 
+  const isInitialLoading = loading && previousData == null;
+
+  const isEmpty = !isInitialLoading && data?.roles?.totalCount === 0;
+
+  const searchBoxProps: ISearchBoxProps = useMemo(() => {
+    return {
+      className: isEmpty ? cn(styles.searchBox, "hidden") : styles.searchBox,
+      placeholder: renderToString("search"),
+      value: searchKeyword,
+      onChange: onChangeSearchKeyword,
+      onClear: onClearSearchKeyword,
+    };
+  }, [
+    isEmpty,
+    renderToString,
+    searchKeyword,
+    onChangeSearchKeyword,
+    onClearSearchKeyword,
+  ]);
+
   const items = useMemo(() => {
     return [{ to: ".", label: <FormattedMessage id="RolesScreen.title" /> }];
   }, []);
 
-  const isEmpty = !loading && data?.roles?.edges?.length === 0;
-
   return (
     <ScreenContent className={styles.content} layout="list">
       <div className={styles.widget}>
-        <NavBreadcrumb className="block" items={items} />
+        <div className={styles.titleContainer}>
+          <NavBreadcrumb className="block" items={items} />
+          {!isEmpty ? <CreateRoleButton /> : null}
+        </div>
+        <SearchBox {...searchBoxProps} />
       </div>
       {isEmpty ? (
         <RolesScreenEmptyState className={styles.widget} />
       ) : (
-        // TODO: implement list view
-        <RolesScreenEmptyState className={styles.widget} />
+        <RolesList
+          className={styles.widget}
```

Comment:
To fix the scrollbar problem, it seems you need min-height: 0 here

Reply:
overflow: auto seems more intuitive to me